### ISSUE TITLE
assert.EventuallyWithT: CollectT is not protected against concurrent use

### ISSUE THREAD

**Issue by dolmen** [stretchr/testify#1418]
In `assert.EventuallyWithT`, if the tick is small, the calls to the condition may overlap (run simultaneously). However they share an access to an `assert.CollectT` instance which is not protected for concurrent use. Race condition may occur.

**Comment by dolmen** [comment]
Cc: @brackendawson

**Comment by dolmen** [comment]
I wonder if we should allow to run the condition function in parallel (and skip ticks) or if instead we should fix error collection to have separate instance for each call.

I have no idea how `EventuallyWithT` is used in the wild.

**Comment by dolmen** [comment]
There is already a pending MR that aims to fix this: #1395

**Comment by dolmen** [comment]
Duplicate of #1391.

**Comment by dolmen** [comment]
#1396 is also related, but not a duplicate.

**Comment by cszczepaniak** [comment]
https://github.com/stretchr/testify/pull/1438 proposes a solution to this that does not require adding any lock objects (i.e. `assert.CollectT` doesn't actually have to be threadsafe to solve this problem)

**Comment by dolmen** [comment]
#1395 is a way to fix this.

**Comment by pmalek** [comment]
Isn't this fixed, given that #1395 has been merged and released?

**Comment by brackendawson** [comment]
I agree, each iteration runs with a new CollectT instance now, thanks @pmalek.
